### PR TITLE
feature(hpa): beta graduation for the container resource metrics

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind api-change
/kind feature

#### What this PR does / why we need it:

graduate the container resource metrics feature on HPA to beta.

#### Which issue(s) this PR fixes:

Related https://github.com/kubernetes/enhancements/issues/1610
Part of https://github.com/kubernetes/kubernetes/issues/115467

#### Special notes for your reviewer:

Required PRs for the beta graduation. We should merge this PR after merging them.

- https://github.com/kubernetes/kubernetes/pull/116010
- https://github.com/kubernetes/kubernetes/pull/116043
- https://github.com/kubernetes/kubernetes/pull/116045
- https://github.com/kubernetes/kubernetes/pull/116326

#### Does this PR introduce a user-facing change?

```release-note
graduate the container resource metrics feature on HPA to beta.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -314,8 +314,9 @@ const (
 	// Make the kubelet use shutdown configuration based on pod priority values for graceful shutdown.
 	GracefulNodeShutdownBasedOnPodPriority featuregate.Feature = "GracefulNodeShutdownBasedOnPodPriority"
 
-	// owner: @arjunrn @mwielgus @josephburnett
+	// owner: @arjunrn @mwielgus @josephburnett @sanposhiho
```
Comment (gjtempleton) [inline pkg/features/kube_features.go:317]: Can we add the link to the KEP in whilst we're here, as per: https://github.com/kubernetes/kubernetes/blob/067b8a80255de4e1d6d51c6eae525c1e673eb01c/pkg/features/kube_features.go#L30

**Comment by sanposhiho** [comment]
/hold

All other issues listed in https://github.com/kubernetes/kubernetes/issues/115467 should be merged before

File: pkg/features/kube_features.go (hunk 1 above)
Comment (sanposhiho) [reply pkg/features/kube_features.go:317]: 👍 
Fixed.

**Comment by sanposhiho** [comment]
/assign @mwielgus 

For approval

**Comment by sanposhiho** [comment]
Now, the last required PR https://github.com/kubernetes/kubernetes/pull/116326 is coming to the merge queue. It's gonna be in the v1.27. 

/unhold

**Comment by sanposhiho** [comment]
The release team said the container metric feature may not be able to in v1.27 
Discussion is on-going in:
https://kubernetes.slack.com/archives/CJH2GBF7Y/p1678843008739849

I'll `/unhold` if https://github.com/kubernetes/kubernetes/pull/116326 can be merged in v1.27

**Comment by marosset** [comment]
/milestone v1.27
/hold cancel

https://github.com/kubernetes/enhancements/issues/1610#issuecomment-1470355665

**Pull request merged**